test/unit_test/loops/unit_test_cpu_loops.cpp

```diff
@@ -0,0 +1,113 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright 2019-2024 Lawrence Livermore National Security, LLC and other
+// DiHydrogen Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: Apache-2.0
+////////////////////////////////////////////////////////////////////////////////
+
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+
+#include "h2/loops/cpu_loops.hpp"
+#include "../tensor/utils.hpp"
+
+
+using namespace h2;
+
+
+TEMPLATE_LIST_TEST_CASE("CPU elementwise loop works",
+                        "[loops]",
+                        h2::ComputeTypes)
+{
+  using Type = TestType;
+
+  SECTION("Empty buffer")
+  {
+    // No assertions, if this compiles and doesn't segfault, we should
+    // be okay.
+    Type* empty_buf = nullptr;
+    cpu::elementwise_loop([](Type) {}, 0, empty_buf);
+    cpu::elementwise_loop(
+        []() -> Type { return static_cast<Type>(42); }, 0, empty_buf);
+    cpu::elementwise_loop([](Type) -> Type { return static_cast<Type>(42); },
+                          0,
+                          empty_buf,
+                          empty_buf);
+  }
+
+  SECTION("Zero buffers and return")
+  {
+    DeviceBuf<Type, Device::CPU> buf{32};
+    buf.fill(static_cast<Type>(0));
+    const Type val = static_cast<Type>(42);
+    cpu::elementwise_loop([&val]() { return val; }, buf.size, buf.buf);
```

Comment:
```suggestion
    cpu::elementwise_loop([&]() { return val; }, buf.size, buf.buf);
```
This will silence the warning.